File: bot.js
const TelegramBot = require('node-telegram-bot-api');

const BOT_TOKEN = process.env.BOT_TOKEN;
const ADMIN_ID = parseInt(process.env.ADMIN_ID || '1444520038');

if (!BOT_TOKEN) {
  console.error('❌ BOT_TOKEN не задан!');
  process.exit(1);
}

const bot = new TelegramBot(BOT_TOKEN, { polling: true });

console.log('✅ Бот запущен');

// Хранилище: кому отвечать (chatId пользователя -> userId)
const replyTargets = new Map();

// Обработка /start
bot.onText(/\/start(.*)/, (msg, match) => {
  const chatId = msg.chat.id;
  const args = match[1] ? match[1].trim() : '';
  const firstName = msg.from.first_name;

  if (args === 'help') {
    bot.sendMessage(chatId, 
      `👋 Привет, ${firstName}!\n\n` +
      `Чем я могу помочь? Опишите проблему, и я свяжусь с вами.`
    );
  } else {
    bot.sendMessage(chatId, 
      `👋 Добро пожаловать в MyStars Support!\n\n` +
      `Я бот поддержки. Используйте /help для связи.`
    );
  }
});

// Обработка /help
bot.onText(/\/help/, (msg) => {
  const chatId = msg.chat.id;
  bot.sendMessage(chatId,
    `🆘 Нужна помощь?\n\n` +
    `Опишите вашу проблему, и администратор свяжется с вами в ближайшее время.`
  );
});

// Обработка обычных сообщений от пользователей
bot.on('message', (msg) => {
  const chatId = msg.chat.id;
  const text = msg.text;

  // Игнорируем команды
  if (!text || text.startsWith('/')) return;

  const userId = msg.from.id;
  const from = msg.from.username ? `@${msg.from.username}` : userId;

  // Если сообщение от админа
  if (userId === ADMIN_ID) {
    // Проверяем, отвечает ли админ на пересланное сообщение
    if (msg.reply_to_message) {
      const repliedText = msg.reply_to_message.text || '';
      
      // Ищем ID пользователя в пересланном сообщении
      const match = repliedText.match(/📩 Сообщение от (@?\w+) \((\d+)\):/);
      if (match) {
        const targetUserId = parseInt(match[2]);
        bot.sendMessage(targetUserId, `👨‍💻 Ответ поддержки:\n\n${text}`)
          .then(() => {
            bot.sendMessage(chatId, `✅ Ответ отправлен пользователю ${match[1]}`);
          })
          .catch((err) => {
            bot.sendMessage(chatId, `❌ Не удалось отправить ответ: ${err.message}`);
          });
        return;
      }
    }
    
    // Обычное сообщение от админа — игнорируем
    return;
  }

  // Сообщение от обычного пользователя — пересылаем админу
  bot.sendMessage(ADMIN_ID,
    `📩 Сообщение от ${from} (${userId}):\n\n${text}`
  ).then(() => {
    bot.sendMessage(chatId, '✅ Ваше сообщение отправлено. Мы ответим в ближайшее время.');
  }).catch((err) => {
    console.error('Ошибка отправки админу:', err);
    bot.sendMessage(chatId, '❌ Произошла ошибка. Попробуйте позже.');
  });
});

File: bot.py
import os
import logging
from telegram import Update
from telegram.ext import Application, CommandHandler, MessageHandler, filters, ContextTypes

logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

BOT_TOKEN = os.environ.get("BOT_TOKEN", "")
ADMIN_ID = int(os.environ.get("ADMIN_ID", "1444520038"))


async def start(update: Update, context: ContextTypes.DEFAULT_TYPE):
    """Обработка команды /start"""
    args = context.args  # Получаем аргументы после /start
    user = update.effective_user
    
    if args and args[0] == "help":
        await update.message.reply_text(
            f"👋 Привет, {user.first_name}!\n\n"
            f"Чем я могу помочь? Опишите проблему, и я свяжусь с вами."
        )
    else:
        await update.message.reply_text(
            f"👋 Добро пожаловать в MyStars Support!\n\n"
            f"Я бот поддержки. Используйте /help для связи."
        )


async def help_command(update: Update, context: ContextTypes.DEFAULT_TYPE):
    """Команда /help"""
    await update.message.reply_text(
        "🆘 Нужна помощь?\n\n"
        "Опишите вашу проблему, и администратор свяжется с вами в ближайшее время."
    )


async def handle_message(update: Update, context: ContextTypes.DEFAULT_TYPE):
    """Обработка обычных сообщений"""
    user = update.effective_user
    message = update.message.text
    
    # Пересылаем сообщение админу
    try:
        await context.bot.send_message(
            chat_id=ADMIN_ID,
            text=f"📩 Сообщение от @{user.username or user.id}:\n\n{message}"
        )
        await update.message.reply_text("✅ Ваше сообщение отправлено. Мы ответим в ближайшее время.")
    except Exception as e:
        logger.error(f"Ошибка отправки админу: {e}")
        await update.message.reply_text("❌ Произошла ошибка. Попробуйте позже.")


def main():
    """Запуск бота"""
    if not BOT_TOKEN:
        logger.error("❌ BOT_TOKEN не задан!")
        return
    
    app = Application.builder().token(BOT_TOKEN).build()
    
    # Регистрируем обработчики
    app.add_handler(CommandHandler("start", start))
    app.add_handler(CommandHandler("help", help_command))
    app.add_handler(MessageHandler(filters.TEXT & ~filters.COMMAND, handle_message))
    
    logger.info("✅ Бот запущен")
    app.run_polling()


if __name__ == "__main__":
    main()
